oci/spec_opts.go

```diff
@@ -1367,3 +1367,17 @@ func tryReadonlyMounts(mounts []mount.Mount) []mount.Mount {
 	}
 	return mounts
 }
+
+// WithWindowsDevice adds a device exposed to a Windows (WCOW or LCOW) Container
+func WithWindowsDevice(idType, id string) SpecOpts {
```

Comment:
Ok last comment. Since I saw the parsing logic in two places for ctr and WithDevices, should we have a very small overload that does:

WithWindowsDeviceFromPath(devicePath string) that does the `split by ://` work so that we can just reuse? Or not worth it?

Reply:
I'm not super-keen on burning `://` into the `oci` package's behaviour. We happen to have two callers that take a single string as a Windows Device spec, but I hope that the CRI caller will become legacy in favour of a real CRI API type for Windows devices.

I do see the case for fixing the string representation though, so I don't super-object to the idea either.

My other concern is whether `class/` should be supported by that API or not. I don't have a clear answer on that, which makes me leery of the API as a whole.

Reply:
I see. Ok then leave it like this. The API is really the OCI RuntimeSpec which supports it as shown here. The `class/` directive was written by someone.. (likely me) to support a way to pass it in via the Docker cli --device syntax without much thought for anything other than class id's for gpu's to process containers. We dont need to carry that. Docker can easily build the oci spec at this level here: https://github.com/moby/moby/blob/dc8fb8f03ba30a80b036706ef89d4dd156729216/daemon/oci_windows.go#L272

So I wouldnt worry too much about `class/`

Reply:
Yeah, that part of the Docker code is where I planned to submit a PR to support `X://Y` as well as `class/GUID`, since I believe dockerd-cri passes the CRI HostPath through the same code.